src/zero/module/aou.clj

```diff
@@ -1,23 +1,385 @@
+                                           :params_file                 "gs://broad-gotc-test-storage/arrays/HumanExome-12v1-1_A/inputs/7991775143_R01C01/params.txt",
+                                           :gender_cluster_file         "gs://broad-gotc-test-storage/arrays/metadata/HumanExome-12v1-1_A/HumanExomev1_1_gender.egt"}))
+
+(defn make-inputs
+  "Return inputs for AoU Arrays processing in ENVIRONMENT from PER-SAMPLE-INPUTS."
+  [environment per-sample-inputs]
+  (let [inputs (merge references/hg19-arrays-references
+                      #_per-sample
+                      (extract-and-validate-per-sample-inputs per-sample-inputs)
+                      chip-metadata
+                      fingerprinting
+                      genotype-concordance
+                      other-inputs
+                      (env-inputs environment))]
+    (util/prefix-keys inputs :Arrays)))
+
+(defn make-options
+  "Return options for aou arrays pipeline."
+  []
+  {; TODO: add :final_workflow_outputs_dir here
+   ; TODO: add :default_runtime_attributes {:maxRetries 3} here
+   :read_from_cache            true
+   :write_to_cache             true
+   :default_runtime_attributes {:zones "us-central1-a us-central1-b us-central1-c us-central1-f"}})
+
+(defn active-or-done-objects
+  "Query by PRIMARY-VALS to get a set of active or done objects from Cromwell in ENVIRONMENT."
+  [environment {:keys [analysis_version_number chip_well_barcode] :as primary-vals}]
+  (prn (format "%s: querying Cromwell in %s" zero/the-name environment))
+  (let [primary-keys [:analysis_version_number
+                      :chip_well_barcode]
+        md           (partial cromwell/metadata environment)]
+    (letfn [(active? [metadata]
+              (let [cromwell-id                       (metadata :id)
+                    analysis-verion-chip-well-barcode (-> cromwell-id md :inputs
+                                                          (select-keys primary-keys))]
+                (when analysis-verion-chip-well-barcode
+                  (let [found-analysis-version-number (:analysis_version_number analysis-verion-chip-well-barcode)
+                        found-chip-well-barcode       (:chip_well_barcode analysis-verion-chip-well-barcode)]
+                    (when (and (= found-analysis-version-number analysis_version_number)
+                               (= found-chip-well-barcode chip_well_barcode))
+                      analysis-verion-chip-well-barcode)))))]
+      (->> {:label  cromwell-label
+            :status ["On Hold" "Running" "Submitted" "Succeeded"]}
+           (cromwell/query environment)
+           (keep active?)
+           (filter seq)
+           set))))
+
+(comment
+  (active-objects :gotc-dev {:analysis_version_number 1, :chip_well_barcode "7991775143_R01C01"}))
```

Comment:
I'm not sure if you were planning to leave this comment in, but I noticed the fn is called `active-or-done-objects` now!

Reply:
nice catch! Yeah, Let me rename it just in case.